crates/ruff/src/rules/flake8_pyi/rules/simple_defaults.rs

```diff
@@ -523,3 +555,36 @@ pub(crate) fn unassigned_special_variable_in_stub(
         stmt.range(),
     ));
 }
+
+/// PIY026
+pub(crate) fn type_alias_check(checker: &mut Checker, value: &Expr, targets: &[Expr]) {
+    let is_any = |expr: &Expr| {
+        let Expr::Name(ast::ExprName { id, .. }) = expr else {
+            return false;
+        };
+        id == "Any"
+    };
+
+    let target = &targets[0];
+    if is_special_assignment(target, checker.semantic()) {
+        return;
+    }
+    if matches!(value, Expr::Name(_)) && !is_any(value) {
+        return;
+    }
+    if !is_valid_pep_604_union(value) {
+        return;
+    }
+
+    let Expr::Name(ast::ExprName { id, .. }) = target else {
+            return;
+    };
+
+    checker.diagnostics.push(Diagnostic::new(
+        TypeAliasCheck {
+            name: id.to_string(),
+            value: checker.generator().expr(value),
+        },
+        target.range(),
+    ));
+}
```

Comment:
It looks like the original implementation in flake8-pyi has slightly different criteria: https://github.com/PyCQA/flake8-pyi/blob/2a86db8271dc500247a8a69419536240334731cf/pyi.py#L1085.

It looks like they trigger if:

- The value is a subscript.
- The value is a PEP 604-style union.
- The value is `Any`.
- The value is `None`.

It looks like the current implementation here is more permissive. I think we need some tweaks to be more conservative, to avoid false positives. E.g., we should omit _any_ expression that isn't a subscript (like, we should omit all call expressions), but we should allow the constant `None`.

Reply:
The length of the docstring on that flake8-pyi method there is a testament to the number of times we had to revise that check before we got to a place where the balance between false negatives and false positives was where we wanted it to be... There are _still_ a few false positives that flake8-pyi's Y026 produces, but overall I'm pretty happy with it now